GLOSSARY.md

```diff
@@ -0,0 +1,44 @@
+# Parts of Class Names
+
+| Name        | Meaning                                                                                                                                                                                                        |
+| ----------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Abstract... | An incomplete class used for *dynamic* polymorphism. It has pure virtual methods that need to be implemented by subclasses.                                                                                    |
+| Base...     | An incomplete class used for *static* polymorphism. It implements methods shared by all implementations. Mostly, it is used so that we have a non-templated superclass, which can be stored in e.g., a vector. |
```

Comment:
I think what you mean is runtimer or dynamic polymorphism.

Reply:
I see how my explanation is confusing. How about this:

**Abstract...**: An incomplete class used for *dynamic* (i.e., run-time) polymorphism. It has pure virtual methods that need to be implemented by subclasses.

**Base...**: An incomplete class used for *static* (i.e., compile-time) polymorphism. The base class itself is not templated, so that we can store, e.g., a `BaseColumn` in a vector of columns with different types. Base classes can also have abstract methods.

Reply:
Maybe:
`Abstract*`: An incomplete class used for dynamic (i.e., run-time) polymorphism. It is a pure interface consisting only of pure virtual methods.

Reply:
Not true. `AbstractOperator::execute` is implemented.